Refactor code to uniform style

Geweldig opened the pull request:
All javadoc has been changed to adhere to a global style.
All  but one PMD/FindBugs/Checkstyle warnings have been addressed, the one in the ScriptLoader class.

This is the way _I_ think the package structure should look, any and all changes are welcome.

This pull request is way larger than a normal pull request should be, but it's the only way to make the code look uniform.

Closes #207, Closes #208, Closes #209

Geweldig committed:
Fix comments AudioManager

Fixes the comments in the AudioManager.

Geweldig committed:
Fi comments BackgroundMusic

This fixes the comments in BackgroundMusic.java.

Geweldig committed:
Fix SoundType comments

Fixes the comments in SoundType.java.

Geweldig committed:
Fix controller comments

Fixes all comments in the controller package.

Geweldig committed:
Fix FileUtil

Moves the FileUtil to the Util package. Fixes all comments.

Geweldig committed:
Fix logging

Fixes the comments in the logging package

Geweldig committed:
Fix model

Fixes all classes in the model package.

Geweldig committed:
Rework roomIO

Reworks all comments in the roomIO package.

Geweldig committed:
Merge branch 'master' into fix-refactor_to_single_style

Geweldig committed:
Fix util package

This fixes all classes in the util package to use the decided upon
javadoc format.

Geweldig committed:
Merge branch 'fix-refactor_to_single_style' of https://github.com/Taeir/ContextProject-MIGI2 into fix-refactor_to_single_style

Geweldig committed:
Fix webinterface and main

Fixes the webinterface package and the Main class to follow our agreed
upon documentation type.

Geweldig committed:
Move roomIO

Moves the roomIO package to model.

Geweldig committed:
Fix audio tests

Fixes the checkstyle for all tests in the audio package.

Geweldig committed:
Fix last tests

All classes have now been reworked.

Geweldig committed:
Make Checkstyle/PMD/FindBugs happy

Resolves all but one FindBugs warning in the project.

Taeir commented on src/main/java/nl/tudelft/contextproject/audio/BackgroundMusic.java:
```diff
@@ -191,15 +178,11 @@ public synchronized AudioNode getCurrent() {
+
 		if (an.getStatus() == AudioSource.Status.Stopped) {
 			next();
 		}
```
So you removed ALL my comments....
Well ok then....
I guess that means my code is pretty clear and understandable.

Geweldig replied:
All comments can be re-added, but I was of the opinion they didn't add any extra understandability to the code. As I said, I'm fine with readding them.

Taeir replied:
No I think you are correct. All comments you removed stated the obvious.

Taeir commented on src/main/java/nl/tudelft/contextproject/controller/GameState.java:
```diff
@@ -4,29 +4,16 @@
-	/**
-	 * The main game is finished.
-	 */
 	ENDED;
```
I think the above comments were somewhat useful, and don't neccesarily had to be removed.

Taeir commented on src/main/java/nl/tudelft/contextproject/logging/Log.java:
```diff
@@ -15,32 +15,17 @@
-	//Log level for fatal messages
 	public static final Level FATAL = createLogLevel("FATAL", 2000);
-	
-	//Global set of loggers, used to properly close FileHandlers.
```
This comment provides useful information and should be readded.

Taeir commented on src/main/java/nl/tudelft/contextproject/logging/Log.java:
```diff
@@ -15,32 +15,17 @@
-		//Log messages to the console like:
-		//14:59:26 [WARNING] [ENGINE]: There is a problem!
-		//Exception - Stacktrace
 		LogFormatter formatter = new LogFormatter("%1$tH:%1$tM:%1$tS [%3$s] [%2$s]: %4$s%5$s%n");
```
The comments above this provide useful information about the format and should be re-added.

Geweldig replied:
I agree, but it was already explained in line 18:
//The default format for files: "HH:MM:SS [LEVEL]: Message", optionally followed by an exception and its stacktrace.

Taeir replied:
Please note that the formatter for the CONSOLE and the formatter for the FILE are not the same.
Note the [%2$s], that the console does have, and the files do not.

Files log: "TIME [LEVEL]: Message", console logs "TIME [LEVEL] [LOGNAME]: Message"
The files don't need the logname, as each file only contains messages from a single logname.

Taeir commented on src/main/java/nl/tudelft/contextproject/logging/LogFormatter.java:
```diff
@@ -37,17 +37,15 @@ public LogFormatter(String format) {
      */
     @Override
 	public String format(LogRecord record) {
         //Format the message
```
Comment can also be removed :P

Taeir commented on src/main/java/nl/tudelft/contextproject/model/EntityState.java:
```diff
@@ -1,19 +0,0 @@
-	/**
-	 * The entity is new and should be added to the renderer.
-	 */
-	NEW 
```
I think the javadocs for the EntityState should be readded, as they provide additional information about what each state implies (should be removed, should receive updates and should be added).

Taeir replied:
(For others also reviewing this, EntityState was moved to the entities package, and since the diff was so large, git did not see it as a MOVE, but rather as a delete + add. See below for the new EntityState).

Taeir commented on src/main/java/nl/tudelft/contextproject/model/Inventory.java:
```diff
@@ -3,13 +3,15 @@
  */
 public class Inventory {
-	ArrayList<Entity> array; 
+	ArrayList<Entity> array;
```
Perhaps this should be renamed to "pickedUpEntities" (or something like that) while we are at it.

Taeir commented on src/main/java/nl/tudelft/contextproject/model/Inventory.java:
```diff
@@ -57,6 +62,7 @@ public void remove(Entity ent) {
 				}
```
Not relevant for this PR, but I see here that an entity is removed in a for each loop over it, which WILL throw a ConcurrentModificationException.

I will create a PR to fix this after this has been merged (to avoid merge conflicts).

Taeir replied:
See #240.

Taeir commented on src/main/java/nl/tudelft/contextproject/model/Inventory.java:
```diff
@@ -69,51 +75,56 @@ public void remove(Entity ent) {
+	 *
 	 * @return
-	 * 		A bomb from the inventory
+	 * 		a bomb from the inventory if the inventory contains a bomb
```
"if the inventory contains a bomb, returns that bomb. Otherwise returns null"
is probably better.

Taeir commented on src/main/java/nl/tudelft/contextproject/model/Inventory.java:
```diff
@@ -139,14 +153,15 @@ public Key getKey(ColorRGBA color) {
 	 * @return
-	 * 		Current size
+	 * 		the current size of the inventory
 	 */
```
I would suggest changing this to "the current amount of items in the inventory (bombs and keys)."

Geweldig committed:
Readd comments

Readds all comments as requested by @Taeir so far.

Taeir commented on src/main/java/nl/tudelft/contextproject/model/PhysicsObject.java:
```diff
@@ -12,8 +12,9 @@
+	 * 		an object that implements PhysicsControl
 	 */
-	public PhysicsControl getPhysicsObject();
+	PhysicsControl getPhysicsObject();
```
I see you have installed the tool that jurgen showed? We don't have the CheckStyle rule that checks for "unneccesary modifiers" enabled.

While this change is certainly correct, I do think that explicitly adding the modifier is not necessarily a bad thing. For example, Java 9 will add private methods to interfaces, which would make "no modifier = public" less obvious for interfaces.

Nevertheless, I'm just nitpicking about something that is a personal preference. It seems like it's common practice to leave the modifiers out, so I agree with this change.

Geweldig replied:
IntelliJ automatically hates you for using modifiers in interfaces, which is why I removed them.

Geweldig committed:
Fix failing build

Fixes the failing build, readds the Log comment.

Taeir commented on src/main/java/nl/tudelft/contextproject/model/entities/VRPlayer.java:
```diff
@@ -1,261 +1,258 @@
+ */
+public class VRPlayer extends Entity implements ActionListener, PhysicsObject {
+	/**
+	 * Physics interaction constants.
```
Should be a comment, not a javadoc

Taeir commented on src/main/java/nl/tudelft/contextproject/model/entities/VRPlayer.java:
```diff
@@ -1,261 +1,258 @@
+	 */
+	public static final float SIDE_WAY_SPEED_MULTIPLIER = .08f;
+	public static final float STRAIGHT_SPEED_MULTIPLIER = .1f;
+
```
I think that some of the comments above the constants provided additional, useful information, and should be readded.
E.g. PLAYER_GRAVITY "how fast the player accelerates while falling" is useful.

Geweldig committed:
Readd comments VRPlayer

People agreed these comments might be useful to keep.

Taeir commented on src/main/java/nl/tudelft/contextproject/model/entities/VRPlayer.java:
```diff
@@ -1,261 +1,258 @@
+	public static final float PLAYER_RADIUS = .5f;
+	public static final float PLAYER_HEIGHT = 3f;
+	//SHOULD NOT BE CHANGED
+	public static final int PLAYER_AXIS = 1;
```
Rename to PLAYER_GRAVITY_AXIS?

Geweldig committed:
Rename constant

Renames the PLAYER_AXIS to better represent what it does.

Taeir commented on src/main/java/nl/tudelft/contextproject/model/entities/VRPlayer.java:
```diff
@@ -1,261 +1,261 @@
+	 * Set character control.
+	 *
+	 * @param characterControl
+	 * 				control to set
```
Indentation should be fixed here.

Geweldig committed:
Change javadoc to comments

Certain javadoc should be comments, these have been changed.

Taeir commented on src/main/java/nl/tudelft/contextproject/model/entities/VRPlayer.java:
```diff
@@ -1,261 +1,261 @@
+				break;
+			default:
+				break;
+			}
```
closing bracket is not indented properly.

Geweldig committed:
Fix indentation on switch

The last bracket was accidentally also tabbed.

Geweldig committed:
Fix broken indentation

Fixes a double tab in VRPlayer

Taeir commented on src/main/java/nl/tudelft/contextproject/model/level/DrawableFilter.java:
```diff
@@ -15,16 +15,21 @@
-	 * @param hideUnexplored Boolean to indicate if explored tiles should be draw.
+	 *
+	 * @param hideUnexplored
+	 * 		boolean to indicate if explored tiles should be draw
```
"draw" should be "drawn"

Taeir commented on src/main/java/nl/tudelft/contextproject/model/level/DrawableFilter.java:
```diff
@@ -33,25 +38,33 @@ public DrawableFilter(Set<String> entitySet, boolean hideUnexplored) {
+	 * @param e
+	 * 		the entity to add
+	 * @return
+	 * 		true when te entity is added, false if it was already in the list
```
true if the entity was added, false if it was already in the set

Taeir commented on src/main/java/nl/tudelft/contextproject/model/level/GeneratorRoom.java:
```diff
@@ -19,11 +19,11 @@
      * Create a Generator Room.
```
Indentation is suddenly spaces instead of tabs like the other files. Some kind of IDE mess up?

Taeir commented on src/main/java/nl/tudelft/contextproject/model/level/Level.java:
```diff
@@ -77,35 +88,28 @@ public MazeTile getTile(int x, int y) {
 		json.put("width", getWidth());
 		json.put("height", getHeight());
-		
-		//Store all the tiles in a JSONObject.
```
I suggest readding this comment. This comment alone should suffice to explain the entire for loop below it.

Geweldig committed:
Fix indentation GeneatorRoom

For some reason GeneratorRoom was indented with spaces. This is no
longer the case.

Taeir commented on src/main/java/nl/tudelft/contextproject/model/level/Level.java:
```diff
@@ -122,18 +126,15 @@ public JSONObject toExploredWebJSON() {
 			JSONArray jArray = new JSONArray();
 			
 			for (int y = 0; y < row.length; y++) {
-				//We are only interested in explored tiles
```
Should be readded IMO.

Geweldig replied:
I didn't realise I removed this comment as I agree with you completely.

Geweldig committed:
Readd comments level

Taeir commented on src/main/java/nl/tudelft/contextproject/model/level/RandomLevelFactory.java:
```diff
@@ -137,9 +134,6 @@ protected int getRandom(int min, int max) {
 
-		/**
-		 * Try to place the requested amount of rooms.
-		 */
```
Readd comment as normal comment.

Geweldig replied:
Readded, somehow diff is still in date.

Taeir commented on src/main/java/nl/tudelft/contextproject/model/level/RandomLevelFactory.java:
```diff
@@ -303,69 +296,75 @@ protected static void carveCorridorWalls(TileType[][] map) {
 	 * @param allowDuplicates
-	 * 			if the size should be deleted from the list after selection
+	 *		if the size should be deleted from the list after selection
      * @return
```
This line is indented with spaces instead of tab.

Geweldig replied:
It isn't on my IDE/GitHub :/

Taeir replied:
Has been _resolved_

Geweldig committed:
Readd comment RandomLevelFactory

Taeir commented on src/main/java/nl/tudelft/contextproject/model/roomIO/RoomReader.java:
```diff
@@ -1,45 +1,60 @@
 	 */
 	public static void importFile(String folder, MazeTile[][] tiles, Set<Entity> entities, List<Light> lights, int xOffset, int yOffset) throws IOException {
-		try (BufferedReader br = new BufferedReader(new FileReader(getMapFile(folder)))) {
+		try (BufferedReader br = new BufferedReader(new InputStreamReader(new FileInputStream(getMapFile(folder)), "UTF-8"))) {
```
A few things.
1) Why was this changed from a FileReader?
2) Doesn't this throw a CharsetNotFound exception that has to be caught or something?

I think new FileReader(..., StandardCharsets.UTF_8) would be better (something like that).

Taeir replied:
Oh, my mistake, I understand this change now.

Taeir replied:
I suggest changing "UTF-8" to StandardCharsets.UTF_8.

Taeir commented on src/main/java/nl/tudelft/contextproject/util/JSONUtil.java:
```diff
@@ -9,7 +9,7 @@
 import java.nio.file.Files;
```
File has space indentation instead of tabs.

Geweldig replied:
Changed, somehow diff is still in date.

Taeir commented on src/main/java/nl/tudelft/contextproject/util/EntityUtil.java:
```diff
@@ -20,9 +20,9 @@ private EntityUtil() {}
      *
      * @param entity
-     *          the classname of the entity
+     *      the classname of the entity
```
File has spaces indentation instead of tabs. Really strange that we have so many files that have this. Eclipse and IntelliJ should both default to tabs only.

Taeir commented on src/main/java/nl/tudelft/contextproject/util/FileUtil.java:
```diff
@@ -82,13 +82,13 @@ public static File getFile(String location) {
 		if (file.exists()) return file;
 
-		//We need to extract this from the jar.
+		//If does file has not yet been extracted we extract it
```
Old comment was proper english, this one is not xD.

Did you mean "File has not yet been extracted, so we extract it."

Geweldig replied:
Does should have been The, I was probably tired when I wrote this

Taeir commented on src/main/java/nl/tudelft/contextproject/util/FileUtil.java:
```diff
@@ -114,23 +115,22 @@ private static void extractFromJar(String fLocation) {
-					//Outputfile already exists
 					continue;
 				} else if (entry.isDirectory()) {
-					//This is a directory, so ensure that all folders are created
```
Should be readded as "Ensure that all parent folders are created"

Geweldig replied:
Readded, somehow diff is still in date.

Taeir commented on src/main/java/nl/tudelft/contextproject/util/FileUtil.java:
```diff
@@ -114,23 +115,22 @@ private static void extractFromJar(String fLocation) {
 					outputFile.mkdirs();
 					continue;
 				} else if (outputFile.getParentFile() != null) {
-					//Ensure that the parent folder is created
```
Should be readded

Geweldig committed:
Fix charset, Fix tabs

Fixes the charset in RoomReader, fixes the tabs
in EntityUtil and Size.

Taeir commented on src/main/java/nl/tudelft/contextproject/util/Size.java:
```diff
@@ -12,9 +12,9 @@
      * Constructor for creating a "Size" pair.
```
File has spaces for indentation again.

Taeir commented on src/main/java/nl/tudelft/contextproject/webinterface/ClientServlet.java:
```diff
@@ -22,9 +22,7 @@
 	private static final String CONTENT_TYPE_JSON = "text/json";
-	
 	private static final Log LOG = Log.getLog("WebInterface");
-	
```
Newline between static and nonstatic fields should be readded.

private static final Log LOG = ...
NEWLINE
private final transient WebServer server;

Geweldig replied:
Readded, somehow diff is still in date.

Taeir commented on src/main/java/nl/tudelft/contextproject/webinterface/ClientServlet.java:
```diff
@@ -53,47 +49,32 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response) t
-		//Log event
 		LOG.fine("Received POST Request: URI=\"" + request.getRequestURI() + "\", Parameters=" + request.getParameterMap());
-		
-		//Handle the post request differently based on the requested URL.
```
Comment should be readded

Taeir commented on src/main/java/nl/tudelft/contextproject/webinterface/ClientServlet.java:
```diff
@@ -53,47 +49,32 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response) t
 				break;
-
 			default:
-				//Unknown post request, so propagate to superclass
```
Comment should be readded.

Taeir commented on src/main/java/nl/tudelft/contextproject/webinterface/WebServer.java:
```diff
@@ -73,56 +68,37 @@ public int getUniqueClientCount() {
-		
-		//Start the webserver
+
 		server.start();
```
Sometimes, I think obvious comments are okay. Removing all the comments from this code makes it more difficult to read, and harder to understand in my opinion. I think that every separate section (separated by the newlines) should have a comment above it here.

Geweldig committed:
Readd comments, change spaces to tabs

Taeir commented on src/main/java/nl/tudelft/contextproject/webinterface/WebServer.java:
```diff
@@ -332,10 +293,8 @@ public static Cookie findCookie(String name, Cookie[] cookies) {
 	 * 		the newly created cookie
 	 */
 	public static Cookie createCookie(HttpServletRequest request) {
-		//Use the session to generate the Session ID.
```
This comment sneakily explains why we have both a SESSION and a PERSISTENT cookie.

I will add a proper explanation why we use this cookie system in a different PR after this one has been merged.

Geweldig replied:
Good idea, this comment in and of itself doesn't explain that, which is why I removed it.

Taeir replied:
See #241

Taeir commented on src/test/java/nl/tudelft/contextproject/MainTest.java:
```diff
@@ -110,6 +110,7 @@ public void testSetControllerAgain() {
 	/**
 	 * Try to get the current game from a state that does not have a current game.
 	 */
+	@Test
```
Well caught! This explains why coverage for Main was below par.

Taeir commented on src/test/java/nl/tudelft/contextproject/audio/AudioManagerTest.java:
```diff
@@ -39,19 +39,12 @@ public void tearDown() {
 		SoundType.AMBIENT.setGain(2.0f);
-		
-		//The volume should have been updated
 		verify(an, times(1)).setVolume(2.0f);
```
While obvious, I think the comments here do give a step by step overview of what is expected to happen after something is done.

I feel that (at least some) comment should be readded, but I'm curious how other team members think about this.

Taeir replied:
The above comment applies to all the changes in this file.

Geweldig replied:
@Taeir and I can understand each other's preference for these comments, but disagree on whether or not they should be added/kept. Someone else's input would be greatly appreciated.

Geweldig committed:
Readd comments WebServer

Taeir commented on src/test/java/nl/tudelft/contextproject/audio/BackgroundMusicTest.java:
```diff
@@ -20,7 +20,7 @@
 /**
```
My comment on AudioManagerTest also applies here.

Taeir commented on src/test/java/nl/tudelft/contextproject/controller/ControllerTest.java:
```diff
@@ -127,6 +133,7 @@ public void testRemoveDrawable() {
 		c.setRootNode(rn);
 		Geometry geom = mock(Geometry.class);
+
 		Drawable d = new Drawable() {
```
While not necessarily related to this PR, we usually do something like this with a mock.

Taeir commented on src/test/java/nl/tudelft/contextproject/logging/TestLog.java:
```diff
@@ -55,17 +55,11 @@ public void setUp() throws Exception {
 	 * 		the exception logged with the message, or null for no exception
```
My comment for the audio tests also applies here.

Taeir replied:
File should be renamed to LogTest for consistency with the rest.

Taeir commented on src/test/java/nl/tudelft/contextproject/logging/TestLog.java:
```diff
@@ -78,8 +72,7 @@ public void verifyLogged(Level level, String msg, Exception ex) {
 	public void testLog() throws IOException {
 		Log nlog = new Log("hello", true);
-		
-		//There should be a filehandler and a consolehandler.
```
For once my comment doesn't give the same information as the code. This one should be readded, as it explains why we expect 2 instead of 42.

Geweldig replied:
I ran through this too quickly, my apologies.

Taeir commented on src/test/java/nl/tudelft/contextproject/logging/TestLog.java:
```diff
@@ -280,10 +269,8 @@ public void testCreateLogLevel() {
-		//Shutdown the log
 		Log.shutdown();
-		
-		//A file handler is closed, when isLoggable returns false for messages that should otherwise be logged.
```
This comment explains why isLoggable is an appropriate way of testing if SHUTDOWN works, so should be readded.

Without this comment, it would seem like the test is simply for coverage only.

Geweldig committed:
Readd comments TestLog.java

Geweldig committed:
Rename TestLog

Renames TestLog to LogTest to stick to our naming convention.

Taeir commented on src/test/java/nl/tudelft/contextproject/model/entities/EntityTest.java:
```diff
@@ -1,123 +1,126 @@
-	/**
-	 * create a new (clean) entity to test with.
-	 */
-	public void setupEntity() {
```
Adding `@Before` here would make a lot of tests cleaner.

Perhaps this should be done in a separate PR. If people agree, I'm fine with creating that PR.

Taeir replied:
See #243

Taeir commented on src/test/java/nl/tudelft/contextproject/util/JSONUtilTest.java:
```diff
@@ -28,113 +28,113 @@
+	 */
+	@Test
+	public void testEntitiesToJson() {
+		Set<Entity> list = ConcurrentHashMap.newKeySet();
```
Nitpicking: rename list to "entities" or "set".

Geweldig replied:
Fixed!

Taeir commented on src/test/java/nl/tudelft/contextproject/webinterface/ClientServletTest.java:
```diff
@@ -29,6 +29,8 @@
 	private static final String ID1 = "TESTID1";
```
My comment about tests like in AudioManagerTest applies here as well.

Taeir commented on src/test/java/nl/tudelft/contextproject/webinterface/ClientServletTest.java:
```diff
@@ -285,211 +250,184 @@ public void testCheckAuthorized_notAuthorized_json() throws IOException {
+		HttpServletRequest request = createMockedRequest(ID1, ID1, true, false, SET_TEAM);
 		HttpServletResponse response = createMockedResponse();
-		
-		//Simulate that the user is authorized
```
This comment does add relevant information.

Geweldig committed:
Rename list to set

Renames the variable "list" of type set in JSONUtilTest
to "set".

Taeir commented on src/test/java/nl/tudelft/contextproject/webinterface/WebServerTest.java:
```diff
@@ -60,7 +60,6 @@ public void tearDown() throws Exception {
 	 */
```
Again the test comment like for the AudioManagerTest.

Taeir commented on src/test/java/nl/tudelft/contextproject/webinterface/WebTestBase.java:
```diff
@@ -34,13 +34,8 @@
 	 */
```
Again see AudioManagerTest comment.

Taeir commented:
Well, a good 2.5 hours later, I have reviewed everything in this pull request.

First off, I want to thank you for doing this. Fixing indentation, javadocs etc. is a simple task that just requires a lot of time. However, it certainly is a worthwhile task, as it not only (mostly) generalizes the code style, as well as bringing to light a few serious issues in our code (which I have highlighted in my comments, and will create issues for in a sec).

## What I have commented about

Most of the comments I have placed are about one or more of the following categories:
1) Comment was removed, but should be added again, as it provided useful information
2) Something is not fixed that should also be fixed in this PR
3) There is a small mistake in the fix.

I have been discussing my comments live with @BCrlrd, and as such he has already addressed quite a few of them at the moment I'm creating this comment.

## What I disagree with

Then there is something I do not really agree with.
This relates to the following classes:
- AudioManagerTest
- BackgroundMusicTest
- TestLog / LogTest
- ClientServletTest
- WebServerTest
- WebTestBase

In all these tests classes, it can be seen that most (if not all) comments in the tests have been removed. It can also be seen that 95% of these comments state the obvious.

As I have written most of these tests myself, I wanted to explain my reasoning for them. When I write a test, I usually try to create some kind of workflow in the comments:
- I do this
- Now this should have happened
- And this should not have happened.

Normally, when I read a test, I ONLY look at the comments (at first). I expect the comments to basically tell me why the code that is below them is there, basically why that is the way to test whatever is being tested. I think that
`//Start music`
`AudioManager.getInstance().startMusic(new AudioNode(...))`

`//Music should have started`
`verify(audioNode).setState(State.PLAYING)`
is easier to understand with the comments, even though the comments say exactly what the code is doing.

As @BCrlrd do not really agree on this matter, I think that we should discuss this with the entire team, to decide how we should comment tests properly.

## Final Words

Apart from this, I think this should definitely be merged once all mistakes have been addressed.

Taeir commented on src/main/java/nl/tudelft/contextproject/logging/LogFormatter.java:
```diff
@@ -31,23 +31,13 @@
-     * 
-     * @return
-     * 		a String representation of the record, using the format of this LogFormatter.
-     */
```
This javadoc provides much simpler, and different information than the one it overrides, so I think it should stay.

Geweldig replied:
I accidentally deleted the wrong comment because I misinterpreted your comment. My apologies.

Geweldig committed:
Fix comment LogFormatter

Because of a misunderstanding the wrong comment was removed. This has
now been fixed.

Geweldig committed:
Fix typo

Diocruel commented:
Looking great so far! I don't have enough time to review the entire pull request. Good work so far by @BCrlrd for the refactoring and @Taeir for checking and discussing the work.

I would like to add a thing: the roomio package should be in the level package as rooms make up a level.

So changing nl.tudelft.contectproject.model.roomIO to nl.tudelft.contectproject.model.level.roomIO

Geweldig committed:
Move roomIO package

Moves the roomIO package to be included in the level package.

Geweldig commented:
I have come to the conclusion that comments in tests should be there and do explain why the tests test what they say, even if they state the obvious. I'll be adding them back in tomorrow and will be adding them to my own tests from now on.

Geweldig committed:
Readd test comments

This readds the comments to the following classes:
* AudioManagerTest
* BackgroundMusicTest
* LogTest
* ClientServletTest
* WebServerTest
* WebTestBase

Taeir commented on src/test/java/nl/tudelft/contextproject/audio/BackgroundMusicTest.java:
```diff
@@ -115,14 +135,19 @@ public void testStart_notPlaying() {
+		//Reset the mock
 		reset(an);
 
+		//Make the audioNode act as being paused
```
*Playing

Taeir replied:
playing*

Taeir commented:
With the latest commit, Bram and I have agreed upon the comments in the tests.

As far as I'm concerned, this should be merged at the earliest opportunity.

Geweldig committed:
Fix incorrect comment

Changes paused to playing, as this is the state the audioNode should be
in.

Geweldig committed:
Fix comment

A comment was accidentally moved to a different line. This has been
fixed.

Diocruel commented:
I agree with all the changes so far. I think it is good to merge this now! Refactoring the code to a singular style greatly increases the readability of the entire code base.
